sympy/matrices/matrices.py

```diff
@@ -2288,6 +2288,35 @@ def irregular(cls, ntop, *matrices, **kwargs):
     def _handle_creation_inputs(cls, *args, **kwargs):
         """Return the number of rows, cols and flat matrix elements.
 
+        Parameters
+        ==========
+
+        sympify_callback_arguments : Boolean, optional
+            If ``True``, it passes SymPy ``Integer`` type as arguments
+            for the callback function.
+
+            If ``False``, it passes python's ``int`` type as arguments
+            for the callback function.
+
+            It only affects the creation of a matrix from the 3
+            arguments, specified as ``rows, cols, func``.
+
+            Default is ``True``.
+
+        sympify_callback_return : Boolean, optional
+            If ``True``, it sympifies the return from the callback.
+
+            If ``False``, it skips the process.
+
+            This flag is unsafe for use if the callabck returns python
```

Comment:
callabck -> callback
python -> Python

The reason I didn't mention float is that any float that is encountered will just get recast to Float when needed. The only danger (and potentially hard to track down bug) is if ints are used with non-future semantics and truncate during division.

```
This flag is unsafe for use if the callback returns non-native
SymPy types. For example, if Python ``ints``  are returned and not sympified
they may
truncate during computation of the inverse of a matrix. In addition, any
non-native SymPy object will not have the methods that
SymPy expects them to have.
```